src/openrct2/ride/TrackPaint.cpp

```diff
@@ -708,14 +708,17 @@ bool TrackPaintUtilDrawStationCovers2(
         imageOffset += SPR_STATION_COVER_OFFSET_TALL;
     }
 
-    auto imageTemplate = session.TrackColours;
-    auto imageId = imageTemplate.WithIndex(baseImageIndex + imageOffset);
+    auto imageId = ImageId(baseImageIndex + imageOffset);
+    if (stationObject->Flags & StationObjectFlags::hasPrimaryColour)
```

Comment:
to account for the trackcolours being ghost/highlighted you could check for `session.TrackColours.IsRemap()` that will only be set on ghost/highlight and if it is set then construct ImageId from the session.TrackColours.
```
if (session.TrackColours.IsRemap()){
    imageId = imageId.WithRemap(session.TrackColours.GetRemap());
}else{
    if (stationObject->Flags & StationObjectFlags::hasPrimaryColour){
    ....
}
```